### PR TITLE
fix: use the real column names rather than the IDs that astropy generates

### LINKED ISSUE #3124: Vizier query for columns with underscore in name
I am unable to get catalogue queries to work on Vizier for columns which contain an underscore. I know I have done it in the past, so I must be doing something daft now. In fact I have the code that used to work about a year or so ago. It still runs, but simply ignores columns that contain an underscore in the name.

Here are a couple of examples on two different catalogues.

```
from astroquery.vizier import Vizier
from astropy.coordinates import SkyCoord
import astropy.units as u
vizierCatName = "II/336/apass9"
catalogs = Vizier.get_catalogs(Vizier.find_catalogs(vizierCatName).keys())
print(catalogs[0].keys())
v = Vizier(catalog=vizierCatName,columns=['RAJ2000', 'DEJ2000', 'Vmag', 'r_mag'])
cent = SkyCoord(ra=100*u.degree, dec=30*u.degree, frame='fk5')
result = v.query_region(cent, radius=0.05*u.degree)
print(result)
print(result[0])

```

The results say three columns where I asked for four. It includes the Vmag data, but not the r_mag. 

> ['RAJ2000', 'DEJ2000', 'e_RAJ2000', 'e_DEJ2000', 'Field', 'nobs', 'mobs', 'B-V', 'e_B-V', 'Vmag', 'e_Vmag', 'Bmag', 'e_Bmag', 'g_mag', 'e_g_mag', 'r_mag', 'e_r_mag', 'i_mag', 'e_i_mag']
> 
> TableList with 1 tables:
> 	'0:II/336/apass9' with 3 column(s) and 20 row(s) 
> 
>  RAJ2000    DEJ2000    Vmag 
>    deg        deg      mag  
> ---------- ---------- ------
> 100.022444  29.953985 14.110
> 100.033717  29.969932     --
> 100.022465  29.954233     --
> 100.010035  29.953945 15.400
>  99.989656  29.959382 15.368
> 100.006628  29.972751 15.475
> 100.017447  29.977752 15.638
> 100.006698  29.972827     --

And a similar query on Tycho, for which the RA,DEC have underscores in their name.

```
from astroquery.vizier import Vizier
from astropy.coordinates import SkyCoord
import astropy.units as u
vizierCatName = "I/259/tyc2"
catalogs = Vizier.get_catalogs(Vizier.find_catalogs(vizierCatName).keys())
print(catalogs[0].keys())
v = Vizier(catalog=vizierCatName,columns=['RA_ICRS_', 'DE_ICRS_', 'BTmag', 'VTmag'])
cent = SkyCoord(ra=100*u.degree, dec=30*u.degree, frame='fk5')
result = v.query_region(cent, radius=0.1*u.degree)
print(result)
print(result[0])

```

Which returns the following. The table includes only two columns for the two magnitudes. No RA,DEC.

> ['TYC1', 'TYC2', 'TYC3', 'pmRA', 'pmDE', 'BTmag', 'VTmag', 'HIP', 'RA_ICRS_', 'DE_ICRS_']
> 
> TableList with 1 tables:
> 	'0:I/259/tyc2' with 2 column(s) and 4 row(s) 
> 
> BTmag  VTmag 
>  mag    mag  
> ------ ------
> 11.540 10.461
> 12.700 11.887
> 11.625 10.366
> 11.831 10.931
>

### PR DESCRIPTION
**Pull request by ManonMarchand**
Hello astroquery maintainers :slightly_smiling_face: 

This PR fixes #3124

The issue was not in the votable parser in astropy, but in the conversion from votable to astropy table where for vizier the option "use_names_overs_ids" should be preferred, as vizier does not write IDs for its fields but uses names instead. 

The IDs that we saw in the issue report are generated by the votable parser when a field has no ID.

See MRE for this:

```python
from astropy.io.votable.tree import VOTableFile, Resource, TableElement, Field
votable = VOTableFile()
table = TableElement(votable)
resource = Resource()
votable.resources.append(resource)
resource.tables.append(table)
table.fields.extend([Field(votable, name="filename(myname", datatype="char", arraysize="*")])
```

```
WARNING: W03: ?:?:?: W03: Implicitly generating an ID from a name 'filename(myname' -> 'filename_myname' [astropy.io.votable.xmlutil]
```

So switching from ID to name for the names of the columns in the conversion from votable to astropy table did the trick.

This is a bit of a breaking change, as people will start to see the real column names rather than the underscore ones. It also appeared from the narrative docs that this strange option was adding some underscores at the beginning of some column names event when no unconventional characters are present.

Edit: the underscore at the beginning is added when the column name starts with a letter... ~something forbidden by the standard that vizier still does :cold_sweat: (and I don't think we can change this upstream)~ 

Edit bis: no, starting with a number is valid. I was confused by the text of the standard, but from reading the xsd at the bottom, the name should be of type ``xsd:token``, which is defined here: https://www.w3.org/TR/xmlschema11-2/#token 
Looks like the xsd:token can be ``2MASS`` without issues :slightly_smiling_face:

### CODE REVIEW

**Comment by keflavich** [comment]
@ManonMarchand you mean if the column started with a number, an underscore was added, right?

This PR looks good to me.  

I think we should have a little discussion about the breaking nature of the PR - do we need to have an option to support the old column naming scheme?  IMO, no, let's fix the bug and move on, but as a heavy user of Vizier, I have relied on its stability in many places in my own software, and a sudden column name change may require substantial refactoring. 

Could you add a Note to the narrative documentation indicating what this change was, why it happened, and (roughly) when it did - i.e., astroquery 0.4.8 / late 2024?  We don't do this in all cases, but given the subtlety of the change and the long-standing nature of the issue in a broadly-used package, it is warranted.

**Comment by cosmoJFH** [comment]
Just for the records, a similar change was done in the gaia module: https://github.com/astropy/astroquery/pull/2967

**Review by bsipocz — APPROVED** [review]
If Adam is happy with getting this change in from the user's POV and think of it more as a bugfix than a behaviour change, then I'm also OK with merging it now.

**Review by keflavich — APPROVED** [review]

**Comment by keflavich** [comment]
Yes, I'm happy with it - but give @ManonMarchand a chance to evaluate my suggested docs rephrasing before we merge.

File: docs/vizier/vizier.rst [hunk 1]
```diff
@@ -14,6 +14,15 @@ radius. Similar to the VizieR web interface, the queries may be further
...
 
+.. note::
+    In astroquery>=0.4.8 an issue where the column names received from this module did
+    not correspond to the columns of the online catalogs was solved.
+    Namely, columns with special characters like ``r'mag`` were renamed into ``r_mag``
+    and columns starting with a number like ``2MASS`` were prepended with an underscore
+    ``_2MASS``.
+    Now that this is fixed, the column names are the same in VizieR's webpages and in
+    the tables received (for the two examples: you'll see ``r'mag`` and ``2MASS``). 
+
```
[the remark this answers is not in the document]
Comment (ManonMarchand) [reply docs/vizier/vizier.rst]: Yup, that's better, thanks!

**Comment by bsipocz** [comment]
Thank you both!

**Comment by taldcroft** [comment]
FYI this change ended up silently breaking some of our code due to the column name changes. Highlighting this as a breaking API change would have been helpful.

**Pull request merged**